todo-app/src/App.vue

```diff
@@ -51,19 +51,34 @@ function deleteTodo(todoId: number) {
 }
 
 function updateTodo(newTodo: Todo, todoId: number) {
-  const indexToUpdateAt = todos.value.findIndex((obj) => obj.id === todoId);
-  todos.value[indexToUpdateAt].content = newTodo.content;
-  todos.value[indexToUpdateAt].title = newTodo.title;
-  todos.value[indexToUpdateAt].importance = newTodo.importance;
-  todos.value[indexToUpdateAt].isEditing = false;
+  const indexToUpdateAt = todos.value.findIndex((todo) => todo.id === todoId);
+  const todoToUpdate = todos.value[indexToUpdateAt];
+  todoToUpdate.content = newTodo.content;
+  todoToUpdate.title = newTodo.title;
+  todoToUpdate.importance = newTodo.importance;
+  todoToUpdate.isEditing = false;
 }
 
-function setIsEditingTrue(todoId: number){
+function setIsEditingTrue(todoId: number) {
   const indexToUpdateAt = todos.value.findIndex((obj) => obj.id === todoId);
   todos.value[indexToUpdateAt].isEditing = true;
 }
 
-function toggleTodoCheckedState(todo: Todo) {
-  todo.isChecked = !todo.isChecked;
+function handleCheckboxClick(todoId: number) {
+  const indexToUpdateAt = todos.value.findIndex((todo) => todo.id === todoId);
+  const todoToMakeFloat = todos.value[indexToUpdateAt];
+  todoToMakeFloat.isChecked = !todoToMakeFloat.isChecked;
+  setTimeout(() => {
+    if (todoToMakeFloat.isChecked === true) {
```

Comment:
You can delete the `=== true` part from the condition,  if the  value of the todoToMakeFloat.isChecked is true, the condition will be true as well. So the result will be the same.